### DISCUSSION TITLE
Supabase File encryption

### DISCUSSION THREAD

**Discussion by PeerC** [supabase/supabase#30611] [Questions]
Hey I'm a pretty new developer who is building his first app and my first question might seem nooby.
I'm building my app with Nuxt and supabase where users can upload video files and images to collaborate on them. Kind of similar to frame.io
Supabase integration was a breeze and I can upload files to the server through my app.

However here is my question:

Is it common practice, that developers can access and view the files, that a user uploads? It doesn't feel right to be able to view the private files through supabase dashboard.
Is there away to prevent this behavior and make the data only accessible for the user who uploaded the file?

Cheers!

**Comment by GaryAustin1** [comment]
There is nothing in Supabase to do this.  

Any system you use where you don't have the encryption key yourself allows the host of the data to look at your data if they really want to.  If they have the encryption key then they can look at the data.  It is just a matter of how easy it is to do that.

**Reply by PeerC** [reply]
Interesting, I thought there must be a way to prevent admins from viewing user files.
